functable.h

```diff
@@ -8,12 +8,18 @@
 
 #include "deflate.h"
 
+typedef void     (* fill_window_cb) (deflate_state *s);
+typedef uint32_t (* update_hash_cb) (deflate_state *const s, uint32_t hash, uint32_t val);
+typedef uint32_t (* adler32_cb)     (uint32_t adler, const unsigned char *buf, size_t len);
+typedef uint32_t (* crc32_cb)       (uint32_t crc, const unsigned char *buf, uint64_t len);
+typedef void     (* slide_hash_cb)  (deflate_state *s);
+
 struct functable_s {
-    void     (* fill_window)    (deflate_state *s);
-    Pos      (* insert_string)  (deflate_state *const s, const Pos str, unsigned int count);
-    uint32_t (* adler32)        (uint32_t adler, const unsigned char *buf, size_t len);
-    uint32_t (* crc32)          (uint32_t crc, const unsigned char *buf, uint64_t len);
-    void     (* slide_hash)     (deflate_state *s);
+    fill_window_cb  fill_window;
+    update_hash_cb  update_hash;
+    adler32_cb      adler32;
+    crc32_cb        crc32;
+    slide_hash_cb   slide_hash;
```

Comment:
This should be a separate PR I think, I don't see any added value from typedef'ing this, it just moves the complexity a few lines and adds a new reference to that complexity. Nobody should be touching this struct unless they also want to touch that complexity, so I don't see the point of this honestly. Feel free to open a new PR with this where you can state the benefit and we can discuss it separately. :)

Reply:
The `update_hash_cb` typedef is used in `insert_string` so there is no indirect reference in the loop.

Reply:
Ah, there it is, I completely missed that. :)

Reply:
Actually it might not be necessary, because indirect lookup doesn't occur. Since it is not a pointer, there only a struct field reference. _I think_.